src/nested_pandas/nestedframe/core.py

```diff
@@ -379,22 +379,32 @@ def my_sum(col1, col2):
         if len(requested_columns) < len(args):
             extra_args = args[len(requested_columns) :]
 
+        # find targeted layers
+        layers = np.unique([col[0] for col in requested_columns])
+
+        # build a flat dataframe with array columns to apply to the function
+        apply_df = NestedFrame()
+        for layer in layers:
+            if layer == "base":
+                columns = [col[1] for col in requested_columns if col[0] == layer]
+                apply_df = apply_df.join(self[columns], how="right")
```

Comment:
I think we should use an "inner" join here to ensure we a) always have a result index that matches the base layer index and b) the order in whi
ch we join doesn't matter.

For example, with `reduce(func, "packed2.e", "packed.c")` on the following:

<img width="555" alt="Screenshot 2024-04-23 at 10 35 59 AM" src="https://github.com/lincc-frameworks/nested-pandas/assets/22227694/8dbcbe2f-b975-4439-b14e-a899e9f93979">

depending on the order of the join, we'll either have a result with no values for index 0 or no values for index 1.  And in this specific cases because `np.unique` returns the unique values in sorted order, we apply the join in alphabetical order of the base layers (so `join(packed, packed2, how="right")` rather than the order in which the columns were specified).

Reply:
Correction, I edited the above to be "outer" rather than "inner" join